lib/model/folder.go

```diff
@@ -88,3 +91,28 @@ func (f *folder) scanTimerFired() {
 
 	f.scan.Reschedule()
 }
+
+func (f *folder) startWatcher() {
+	ctx, cancel := context.WithCancel(f.ctx)
+	f.model.fmut.RLock()
+	ignores := f.model.folderIgnores[f.folderID]
+	f.model.fmut.RUnlock()
+	eventChan, err := f.Filesystem().Watch(".", ignores, ctx, f.IgnorePerms)
+	if err != nil {
+		l.Warnf("Failed to start filesystem watcher for folder %s: %v", f.Description(), err)
+	} else {
+		f.watchChan = make(chan []string)
+		f.watchCancel = cancel
```

Comment:
I suspect this needs locking, as start can be called with ignores updated nearby I guess.

Reply:
It is now only called from the "main loop", so no racyness possible anymore.